src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -115,33 +115,112 @@ public static void Analyze(CSharpCompilation compilation, Symbol member, BoundNo
 
         protected override void Normalize(ref LocalState state)
         {
+            int oldNext = state.Capacity;
             state.EnsureCapacity(nextVariableSlot);
+            for (int slot = oldNext; slot < nextVariableSlot; slot++)
+            {
+                var value = GetDefaultState(ref state, slot);
+                state[slot] = value;
+            }
+        }
+
+        private bool? GetDefaultState(ref LocalState state, int slot)
+        {
+            if (slot == 0)
+            {
+                return null;
+            }
+
+            var variable = variableBySlot[slot];
+            var symbol = variable.Symbol;
+
+            switch (symbol.Kind)
+            {
+                case SymbolKind.Local:
+                    return null;
+                case SymbolKind.Parameter:
+                    {
+                        var parameter = (ParameterSymbol)symbol;
+                        return (parameter.RefKind == RefKind.Out) ?
+                            null :
+                            !parameter.Type.IsNullable;
+                    }
+                case SymbolKind.Field:
+                case SymbolKind.Property:
+                    {
+                        int containingSlot = variable.ContainingSlot;
+                        if (variableBySlot[containingSlot].Symbol.GetTypeOrReturnType().TypeKind == TypeKind.Struct &&
+                            state[containingSlot] == null)
```

Comment:
>state[containingSlot] == null [](start = 28, length = 29)

I am not sure I understand the significance of this condition. Could you please elaborate why the state for containing structure is important? Would it ever be not null?

Reply:
The state for the containing struct is not null if that container is a field in a tracked value. For instance, `G` in `c.G.F` below:
```
struct S
{
    internal object? F;
}
class C
{
    S G;
    static void Main()
    {
        C c = new C();
        c.G.F.ToString(); // warning dereference of null c.G.F
    }
}
```